astropy/io/fits/hdu/base.py

```diff
@@ -1311,55 +1285,39 @@ def verify_checksum(self, blocking='standard'):
 
         if 'CHECKSUM' in self._header:
             if 'DATASUM' in self._header:
-                datasum = self._calculate_datasum(blocking)
+                datasum = self._calculate_datasum()
             else:
                 datasum = 0
-            checksum = self._calculate_checksum(datasum, blocking)
+            checksum = self._calculate_checksum(datasum)
             if checksum == self._header['CHECKSUM']:
                 return 1
-            elif blocking == 'either':
-                # i.e. standard failed,  try nonstandard
-                return self.verify_checksum(blocking='nonstandard')
             else:
-                # Failed with all permitted blocking kinds
+                # Failed
                 return 0
         else:
             return 2
 
-    def _verify_checksum_datasum(self, blocking):
+    def _verify_checksum_datasum(self):
         """
         Verify the checksum/datasum values if the cards exist in the header.
         Simply displays warnings if either the checksum or datasum don't match.
         """
 
-        # NOTE:  private data members _checksum and _datasum are
-        # used by the utility script "fitscheck" to detect missing
-        # checksums.
-
         if 'CHECKSUM' in self._header:
```

Comment:
Not a big problem, but it would be nice to avoid doing this check more than once since it also happens in `verify_checksum`. This could potentially be refactored since `verify_checksum` will return `2` if the `'CHECKSUM'` keyword is missing. Same goes for the check for `'DATASUM'` below.